azurerm/internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -0,0 +1,359 @@
+			  "type": "String"
+		  }
+	  },
+	  "variables": {},
+	  "resources": [
+		  {
+			  "type": "Microsoft.Logic/workflows",
+			  "apiVersion": "2017-07-01",
+			  "name": "[parameters('workflows_Test_name')]",
+			  "location": "${azurerm_resource_group.test.location}",
+			  "properties": {
+				  "state": "Enabled",
+				  "definition": {
+					  "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
+					  "contentVersion": "1.0.0.0",
+					  "parameters": {
+						  "$connections": {
+							  "defaultValue": {},
+							  "type": "Object"
+						  }
+					  },
+					  "triggers": {
+						  "When_Azure_Sentinel_incident_creation_rule_was_triggered": {
+							  "type": "ApiConnectionWebhook",
+							  "inputs": {
+								  "body": {
+									  "callback_url": "@{listCallbackUrl()}"
+								  },
+								  "host": {
+									  "connection": {
+										  "name": "@parameters('$connections')['azuresentinel']['connectionId']"
+									  }
+								  },
+								  "path": "/incident-creation"
+							  }
+						  }
+					  },
+					  "outputs": {}
+				  },
+				  "parameters": {
+					  "$connections": {
+						  "value": {
+							  "azuresentinel": {
+								  "connectionId": "[parameters('connections_azuresentinel_externalid')]",
+								  "connectionName": "azuresentinel",
+								  "id": "/subscriptions/${data.azurerm_client_config.current.subscription_id}/providers/Microsoft.Web/locations/${azurerm_resource_group.test.location}/managedApis/azuresentinel"
+							  }
+						  }
+					  }
+				  }
+			  }
+		  }
+	  ]
+  }
+  TEMPLATE
+
+  depends_on = [azurerm_template_deployment.testconnection]
+}
+
+resource "azurerm_sentinel_automation_rule" "test" {
+  name                       = "%s"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "acctest-SentinelAutoRule-%d-update"
+  order                      = 2
+  enabled                    = false
+  condition {
+    property = "IncidentTitle"
+    operator = "Contains"
+    values   = ["a", "b"]
+  }
+
+  condition {
+    property = "IncidentTitle"
+    operator = "Contains"
+    values   = ["c", "d"]
+  }
+
+  action_incident {
+    order                  = 1
+    status                 = "Closed"
+    classification         = "BenignPositive_SuspiciousButExpected"
+    classification_comment = "whatever reason"
+  }
+
+  action_incident {
+    order  = 3
+    labels = ["foo", "bar"]
+  }
+
+  action_incident {
+    order    = 2
+    severity = "High"
+  }
+
+  action_incident {
+    order    = 4
+    owner_id = data.azurerm_client_config.current.object_id
+  }
+
+  action_playbook {
+    order        = 5
+    logic_app_id = "/subscriptions/${data.azurerm_client_config.current.subscription_id}/resourceGroups/${azurerm_resource_group.test.name}/providers/Microsoft.Logic/workflows/Test"
```

Comment:
given we don't yet have resources to support this maybe we remove these properties until they exist?

Reply:
Now that it works actually, which means the functionality of this property (block) has no problem. Also some customers (e.g. @kaovd) exactly want this.

Reply:
Agree with this, while current workaround may use an arm template to test this, it works fully. Having this feature is an absolute game changer for everyone with sentinel deployments. I dont see why we should throw this out the window just because of how tests are implemented